### ISSUE TITLE
Distribution installation fails with error 0x8007019e on `windows-2025` preview image

### ISSUE THREAD

**Issue by ssbarnea** [Vampire/setup-wsl#65]
As github is always introducing changes to their default github runners, it would prove very useful to document their support, so users will not waste time trying to use this action on machines where it would not work.

For example `windows-2025` is now on public preview and it seems that we cannot use the action on it. See https://github.com/pycontribs/mise-en-gha/actions/runs/12535474178/job/34957424199?pr=1

**Comment by Vampire** [comment]
Actually your first sentence already is the perfect argument against such a support matrix.
They constantly change things, sometimes breaking things.
Actually, the support matrix is there in the readme.
It states that `windows-2019` and newer is supported for WSLv1 and `windows-2022` and newer for WSLv2.
If you use a non-windows runner you get a clean and meaningful error from the action.
If you use WSLv2 on `windows-2019` you get a clean and meaningful error from the action.

There are also daily running integration tests that test all functionality on a variety of combinations and on all runners,
so usually I should also relatively fast get aware of problems GitHub introduced recently and can either report, adapt, or work-around it.

Of course a new runner is not yet part of the test matrix, and especially as it is in preview. (Not that I was aware of it yet at all. :-D )

**Comment by ssbarnea** [comment]
> The problem is, that WSL is not installed yet in the current `windows-2025` image: [actions/runner-images#11265](https://github.com/actions/runner-images/issues/11265). It gets added with this recently merged PR: [actions/runner-images#11242](https://github.com/actions/runner-images/pull/11242).
> 
> But I can probably work-around it in this action. Currently tests are running, if they run through, I'll be able to release a new version of the action that compensates the problem. In the meantime you could call `wsl --install --no-distribution` before using the action, then it should also work already on `windows-2025`.

Thanks again! The reason why I raise it because when they update the runners is high likely that they will break stuff like WSL but if we catch there issues while the new runners are still in preview, we have a very good chance of getting a fix in quickly. After the release,... it can take months.

PS. There is a notable difference between having a specific runner image updated in time and having totally new runner image. Usually updates do not break things, but switching from ubuntu-22.04 to ubuntu-24.04 was a BIG change (same for other operating systems).

After years of using `-latest` runners, I no long do this, as it causes too many issues and is hard to debug. Also the random date when they switch to newer version makes them quite unreliable.

**Comment by Vampire** [comment]
> Thanks again! The reason why I raise it because when they update the runners is high likely that they will break stuff like WSL

Yes, that's why each night a full test run of all functionality on all images is done.
So usually I get a notification as soon as they break something.

> if we catch there issues while the new runners are still in preview, we have a very good chance of getting a fix in quickly

Yes, definitely.
That's why I right away added `windows-2025` to the test matrix, worked-around the problems, and also created an issue for the problem.
So you are very welcome to create a ticket or PR that adds new runners as soon as they are publicly available like with this ticket.

> Usually updates do not break things

"Usually", lol.
If it just were that way. :-D
They also broke existing runners in the past multiple times.
